content/technotes/2020-04-30-rmd-citations/index.Rmd

```diff
@@ -0,0 +1,290 @@
+    fmtPrefix = function(paper) paste0("[^", attr(paper,"key"), "]:"),
+    extraInfo = function(paper) paste0(paper$url, get_doi(paper)),
+    .init = TRUE)
+RefManageR::PrintBibliography(mybib, .opts = list(bib.style = "apa", sorting = ""))
+```
+
+#### Workflow summary
+
+To summarize our workflow
+
+* Instead of referencing the bibliography file in the Document metadata we create an object refering to it in a chunk by calling `RefManageR::ReadBib()`;
+
+* We use a custom-made `cite()` function using `RefManageR::NoCite()` to signal the use of the reference, and string manipulation to add it in the correct format;
+
+* We make the bibliography appear using `RefManageR::PrintBibliography` and `tools::bibstyle()` in a chunk with the `results="asis"` option, chunk that we put exactly where we want the bibliography to appear.
+
+### Do even more with BibTeX files
+
+The workflow presented earlier was a cool example of using RefManageR to use a BibTeX file.
+There are other two packages that are worth knowing about for more .bib gymnastics: bib2df and handlr.
+
+#### bib2df: from BibTeX to tibble
+
+[bib2df](https://docs.ropensci.org/bib2df/) by [Philipp Ottolinger](http://www.ottlngr.de/)[^social] is a package that converts bibliography data from .bib to `tibble` and vice versa. 
+Like RefManageR it has also been [peer-reviewed by rOpenSci](https://github.com/ropensci/software-review/issues/124).
+In the [words of one the reviewers, Adam Sparks](https://github.com/ropensci/software-review/issues/124#issuecomment-308617830), bib2df is _"something that is simple and does one job and does it well."_.
+
+Let's try it on our .bib file.
+
+```{r bib2df}
+df <- bib2df::bib2df("refs.bib")
+df
+bib2df::df2bib(df)
+```
+
+Note that RefManageR has a similar export function.
+
+
+```{r coerce}
+library("magrittr")
+df2 <- "refs.bib" %>%
+  RefManageR::ReadBib() %>%
+  as.data.frame()
+df2
+```
+
+A difference in the formats is for instance the way authors are parsed.
+
+```{r authors}
+df$AUTHOR
+df2$author
+```
+
+`bib2df` even supports separating name, using [`humaniformat`](https://github.com/ironholds/humaniformat/)
+
+```{r humani}
+bib2df::bib2df("refs.bib", separate_names = TRUE)$AUTHOR
+```
+
+bib2df helps doing fun or serious analyses or reference databases.
+
+<!--html_preserve-->
+{{< tweet 887962776806842369 >}}
+<!--/html_preserve-->
+
+#### handlr: from BibTeX to RIS, schema.org, etc.
+
+[handlr](https://docs.ropensci.org/handlr/) by [Scott Chamberlain](/author/scott-chamberlain) is less mature but not less cool.
+It's _a tool for converting among citation formats_, inspired by Ruby [bolognese library](https://github.com/datacite/bolognese).
+It has an all-in-one objects, but you could also use [individual functions reading and writing different formats](https://docs.ropensci.org/handlr/reference/index.html).
+
+```{r handlr}
+citation <- handlr::HandlrClient$new(x = "refs.bib")
+citation
+citation$write(format = "citeproc")
```

Comment:
if you want a more brief output than what `citeproc` produces you could use `citation$write(format = "ris")`